linkerd/app/outbound/src/sidecar.rs

```diff
@@ -74,15 +90,92 @@ impl Outbound<()> {
 
                     http::Routes::Endpoint(Remote(ServerAddr(*orig_dst)), Default::default())
                 };
-                let routes = match (*parent).profile.clone() {
-                    Some(profile) => {
-                        let mut rx = watch::Receiver::from(profile);
-                        let init = mk_routes(&*rx.borrow_and_update());
-                        http::spawn_routes(rx, init, move |profile: &profiles::Profile| {
-                            Some(mk_routes(profile))
+
+                let mk_policy_routes = move |policy: &policy::ClientPolicy| {
+                    // A `ClientPolicy` does not have a single known logical
+                    // address, as it may consist of multiple distinct backends.
+                    // Therefore, just use the original destination address for now.
+                    // TODO(eliza): eventually, remove the logical addr field on
+                    // the policy route types when we use metadata for those
+                    // labels instead.
+                    let addr = orig_dst.0.into();
```

Comment:
Should we extract above instead? I.e. `let OrigDstAddr(orig_dst) = (*param).param();` so we don't have to do awkward `.0` stuff?

Reply:
elsewhere we also use that addr as an `OrigDstAddr`, so we would need to
re-wrap it in `OrigDstAddr` in those places. which is probably fine?

alternatively, we could just change this into 
```suggestion
                    let OrigDstAddr(addr) = orig_dst;
```
which is equivalent but IMO less ugly...